common/lib/network-utils.ts

```diff
@@ -0,0 +1,20 @@
+import dns from 'dns/promises';
+
+/**
+ * Check if the system has internet connectivity by testing DNS resolution.
+ *
+ * @returns Promise that resolves to true if online, false if offline
+ */
+export async function isOnline(): Promise< boolean > {
+	try {
+		const timeoutPromise = new Promise( ( _, reject ) =>
+			setTimeout( () => reject( new Error( 'Timeout' ) ), 5000 )
+		);
+
+		await Promise.race( [ dns.resolve( 'google.com' ), timeoutPromise ] );
```

Comment:
This was the simplest suggestion from our best friend Claudio. But happy to switch to any other context agnostic approach.

Reply:
This approach looks good to me. I would suggest changing the domain to our public-api.

```suggestion
		await Promise.race( [ dns.resolve( 'public-api.wordpress.com' ), timeoutPromise ] );
```